### PR TITLE
DYN- 6893 input and output nodes should not show in home workspace.

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

see original PR here for more info:
https://github.com/DynamoDS/Dynamo/pull/15204

Screenshot 2024-05-28 at 1 04 25 PM
Screenshot 2024-05-28 at 1 04 22 PM

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

😢 - AFAIK this will need to be tested with integration tests - @avidit do we have any tests that do image comparisons or html comparisons of the library?

### Release Notes

fix a legacy issue where input and output symbol nodes were shown in the HomeWorkspace context even though they are only useful in CustomNodeWorkspaces.

### CODE REVIEW

**Review by pinzart90 — APPROVED** [review]

**Comment by avidit** [comment]
We did have tests to compare items in the library but due to lack of TestComplete's support for webview2, the tests are not running at the moment.

File: src/LibraryViewExtensionWebView2/ViewExtension.cs [hunk 1]
```diff
@@ -37,7 +38,17 @@ public void Loaded(ViewLoadedParams viewLoadedParams)
...
             }
-            
+            viewParams.CurrentWorkspaceChanged += ViewParams_CurrentWorkspaceChanged;
+        }
+
+        private void ViewParams_CurrentWorkspaceChanged(IWorkspaceModel workspace)
+        {
+            var type = workspace switch
+            {
+                HomeWorkspaceModel _ => "home",
```
Comment (RobertGlobant20) [inline src/LibraryViewExtensionWebView2/ViewExtension.cs:48]: what do you think of having an enum instead of a hard-coded string?

**Review by RobertGlobant20 — APPROVED** [review]
LGTM

File: src/LibraryViewExtensionWebView2/ViewExtension.cs (hunk 1 above)
Comment (mjkkirschner) [reply src/LibraryViewExtensionWebView2/ViewExtension.cs:48]: should be straight forward, will look into it.

**Comment by mjkkirschner** [comment]
> We did have tests to compare items in the library but due to lack of TestComplete's support for webview2, the tests are not running at the moment.

would it be crazy to implement dumping the library to html and then doing our own snapshotting in the meantime?

**Comment by avidit** [comment]
> > We did have tests to compare items in the library but due to lack of TestComplete's support for webview2, the tests are not running at the moment.
> 
> would it be crazy to implement dumping the library to html and then doing our own snapshotting in the meantime?

We should do something like that as we are not sure when will TestComplete support webview2

**Pull request merged**